### PR TITLE
refactor(new layout) phase one

### PR DESCRIPTION
**Pull request by srirachy**
## Changes
1. Revamping layout
2. Setup for multiple themes (Spotify Clone, Discover, and Music Visualizer Themes)
3. Transitioning some useStates to Redux

## Purpose
Planning to have multiple theme layouts to follow a Spotify Clone theme, a Discover theme which will basically follow closer to my layout from wireframe.cc, and Music Visualizer theme to show off that feature once its done.

## Approach
Basically transitioning from tutorial layout to a more fleshed out layout before wrapping up on the control buttons (shuffle/loop).

## Learning
(https://javascript.plainenglish.io/accessing-the-spotifyalgorithm-full-stack-application-tutorial-part-1-792a2c0ff13)

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -1,147 +1,30 @@
...
-  const [token, setToken] = useState('');
+  const dispatch = useAppDispatch();
+  const token = useSessToken();
 
   useEffect(() => {
     const getToken = async () => {
       const response = await fetch('/auth/token');
       const data = await response.json();
-      setToken(data.access_token);
+      dispatch(setSessToken(data.access_token));
```
Comment (frankieavina) [inline client/src/App.tsx:18]: Don't know exactly how you're storing your token but you should look into storing in browser local storage  provides persistence across page refreshes and browser tabs. You can get XSS attacks but you get lower the risks by configuring your tokens to expire after a short period of time (eg. 1 hour).

File: client/src/components/MusicPlayer/MusicPlayer.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// this component will house the three themes: clone, my own style - emphasis on music discovery, and visualizer
```
Comment (frankieavina) [inline client/src/components/MusicPlayer/MusicPlayer.tsx:1]: Excited to see how this will look !

**Review by frankieavina — APPROVED** [review]
Great progress keep it up. Side note: Javi still owes us lunch and a lap dance

File: client/src/App.tsx (hunk 1 above)
Comment (srirachy) [reply client/src/App.tsx:18]: I get the token through authorization from a spotify url and store it in a redux global state for now. The token expire is pretty much built-in from the spotify api already. Hopefully this link sheds light on the token info: https://developer.spotify.com/documentation/general/guides/authorization/code-flow/

I think your point on having token to persist across page refresh and browser tab is good to bring up though and I'm not entirely sure if a global state from redux solves it yet. If it does not, then I'll for sure look into storing it in local storage. It seems to be fine at its current state though.
